routes/api/v1.php

```diff
@@ -26,12 +28,17 @@ function () {
             Route::patch('/media/{media}/version/{version}', [VersionController::class, 'setVersion'])->name('setVersion');
 
             // Image
-            Route::get(sprintf('/%s/{media}/version/{version}/original', MediaType::IMAGE->value), [ImageController::class, 'getOriginal'])->name('getOriginal');
-            Route::get(sprintf('/%s/{media}/version/{version}/derivative/{transformations?}', MediaType::IMAGE->value), [ImageController::class, 'getDerivativeForVersion'])->name('getDerivativeForVersion');
-            Route::post(sprintf('/%s/reserveUploadSlot', MediaType::IMAGE->value), [UploadSlotController::class, 'reserveImageUploadSlot'])->name('reserveImageUploadSlot');
+            Route::get(sprintf('/%s/{media}/version/{version}/original', MediaType::IMAGE->value), [ImageController::class, 'getOriginal'])->name('getImageOriginal');
+            Route::get(sprintf('/%s/{media}/version/{version}/derivative/{transformations?}', MediaType::IMAGE->value), [ImageController::class, 'getDerivativeForVersion'])->name('getImageDerivativeForVersion');
+            Route::post(sprintf('/%s/reserveUploadSlot', MediaType::IMAGE->value), fn(UploadSlotRequest $request) => app(UploadSlotController::class)->reserveUploadSlot($request, MediaType::IMAGE))->name('reserveImageUploadSlot');
+
+            // Document
+            Route::get(sprintf('/%s/{media}/version/{version}/original', MediaType::DOCUMENT->value), [DocumentController::class, 'getOriginal'])->name('getDocumentOriginal');
+            Route::get(sprintf('/%s/{media}/version/{version}/derivative/{transformations?}', MediaType::DOCUMENT->value), [DocumentController::class, 'getDerivativeForVersion'])->name('getDocumentDerivativeForVersion');
+            Route::post(sprintf('/%s/reserveUploadSlot', MediaType::DOCUMENT->value), fn(UploadSlotRequest $request) => app(UploadSlotController::class)->reserveUploadSlot($request, MediaType::DOCUMENT))->name('reserveDocumentUploadSlot');
 
             // Video
-            Route::post(sprintf('/%s/reserveUploadSlot', MediaType::VIDEO->value), [UploadSlotController::class, 'reserveVideoUploadSlot'])->name('reserveVideoUploadSlot');
+            Route::post(sprintf('/%s/reserveUploadSlot', MediaType::VIDEO->value), fn(UploadSlotRequest $request) => app(UploadSlotController::class)->reserveUploadSlot($request, MediaType::VIDEO))->name('reserveVideoUploadSlot');
```

Comment:
reserve upload slot should only be registered once, using the typehinted enum as parameter instead of sprintf-ing every case to a static url and pass the case explicitly in a closure again.

Reply:
Will be done in #88.